src/coreclr/vm/ceeload.cpp

```diff
@@ -478,6 +478,16 @@ void Module::Initialize(AllocMemTracker *pamTracker, LPCWSTR szName)
     m_dwTypeCount = 0;
     m_dwExportedTypeCount = 0;
     m_dwCustomAttributeCount = 0;
+#ifdef PROFILING_SUPPORTED
+    // set profiler related JIT flags
+    if(CORProfilerDisableInlining()){
+        m_dwTransientFlags |= PROF_DISABLE_INLINING;
+    }
+    if(CORProfilerDisableOptimizations()){
+        m_dwTransientFlags |= PROF_DISABLE_OPTIMIZATIONS;
+    }
+#endif //PROFILING_SUPPORTED
```

Comment:
The below `#if defined(PROFILING_SUPPORTED) && !defined(DACCESS_COMPILE)` has a redundant check on `!defined(DACCESS_COMPILE)`. Can we remove the redundant check and group this under the same guard?

Nit to keep style consistent: space after if and `{` on next line

Reply:
To clarify, is the intention that the bits corresponding to PROF_DISABLE_INLINING and PROF_DISABLE_OPTIMIZATIONS will only be set during initialization of the module?
Should we prevent them from being controlled by `SetTransientFlagInterlocked`?

Reply:
I think the debugger bits are in the same category. I'll check how / if it is done and apply the same logic.

Reply:
debugger flags are not protected either (I just checked). SetTransientFlagInterlocked being private, I think it is ok to leave it as it is in that PR and avoid adding more changes to it. Introducing a "DynamicFlagsMask" could be done in a followup.

Reply:
I think its fine that the checks are no stronger than what we do for debugging, ie no additional change needed for this.